File: android-app/app/src/main/java/com/kvnetflix/mobile/data/repository/MovieRepository.kt
package com.kvnetflix.mobile.data.repository

import com.kvnetflix.mobile.data.api.ApiClient
import com.kvnetflix.mobile.data.model.*

class MovieRepository {

    private val api get() = ApiClient.api

    suspend fun getHomeVideos(category: String? = null, page: Int = 1): HomeResponse {
        val list = api.getHomeVideos(category, page)
        return HomeResponse(items = list, totalPages = 10, currentPage = page)
    }

    suspend fun searchVideos(query: String, page: Int = 1): HomeResponse {
        val list = api.searchVideos(query, page)
        return HomeResponse(items = list, totalPages = 1, currentPage = page)
    }

    suspend fun getMovieDetail(slug: String): MovieDetail {
        val response = api.getMovieDetail(slug)
        return MovieDetail(
            id = response.id,
            title = response.title,
            originalTitle = response.originalTitle,
            slug = response.slug,
            thumbnail = response.thumbnail,
            backdrop = response.backdrop,
            quality = response.quality,
            year = response.year,
            category = response.category,
            description = response.description,
            rating = response.rating,
            duration = response.duration,
            genre = response.genre,
            director = response.director,
            country = response.country,
            cast = response.cast,
            episodes = response.episodes
        )
    }

    suspend fun extractVideo(url: String): VideoSource {
        return api.extractVideo(ExtractRequest(url))
    }

    suspend fun getGenres(): List<Category> {
        return api.getGenres()
    }

    suspend fun getCountries(): List<Category> {
        return api.getCountries()
    }

    suspend fun exploreMovies(): List<Movie> {
        return api.exploreMovies()
    }
}

File: ios-app/KvNetflix/Repositories/MovieRepository.swift
import Foundation

public final class MovieRepository {
    public static let shared = MovieRepository()
    private let api = ApiClient.shared

    public init() {}

    public func getHomeVideos(category: String? = nil, page: Int = 1) async throws -> HomeResponse {
        let items = try await api.getHomeVideos(category: category, page: page)
        return HomeResponse(items: items, totalPages: 10, currentPage: page)
    }

    public func searchVideos(query: String, page: Int = 1) async throws -> HomeResponse {
        let items = try await api.searchVideos(query: query, page: page)
        return HomeResponse(items: items, totalPages: 1, currentPage: page)
    }

    public func getMovieDetail(slug: String) async throws -> MovieDetail {
        return try await api.getMovieDetail(slug: slug)
    }

    public func extractVideo(url: String) async throws -> VideoSource {
        return try await api.extractVideo(url: url)
    }

    public func getStreamSubtitles(embedUrl: String) async throws -> [SubtitleTrack] {
        return try await api.getStreamSubtitles(embedUrl: embedUrl)
    }

    public func getGenres() async throws -> [Category] {
        return try await api.getGenres()
    }

    public func getCountries() async throws -> [Category] {
        return try await api.getCountries()
    }

    public func exploreMovies() async throws -> [Movie] {
        return try await api.exploreMovies()
    }
}
